apps/coscrad-frontend-e2e/src/e2e/resource-index-to-detail-flows/spatial-feature.query-flow.e2e.cy.ts

```diff
@@ -0,0 +1,60 @@
+describe(`Spatial Feature index-to-detail flow`, () => {
+    const textForSpatialFeatureMarker = 'Name of Point with ID: 103';
+
+    describe(`the resource menu`, () => {
+        beforeEach(() => {
+            cy.visit(`/Resources`);
+        });
+
+        it('should have an entry for spatial features', () => {
+            // TODO: we should have a human readable resource name and modify this test to look for that.
+            cy.contains('spatialFeatures');
+        });
+
+        it('should have a link to the spatial features', () => {
+            cy.contains('spatialFeatures').click();
+
+            /**
+             * We don't have a title for this page since the map is self-evident.
+             * This test however should not be tied to the implementation of the Leaflet plugin
+             */
+            cy.contains('Leaflet');
+
+            cy.location('pathname').should('contain', 'Resources/Map');
+        });
+    });
+
+    describe(`the spatial feature index page`, () => {
+        beforeEach(() => {
+            cy.visit(`/Resources/Map`);
+        });
+
+        it('should display a leaflet map', () => {
+            // Not sure if this is working - map appears not to move
+            cy.get('.leaflet-container').dragMapFromCenter({
+                xMoveFactor: 60,
+                yMoveFactor: -100,
+            });
+        });
+
+        it(`should display spatial feature markers`, () => {
+            cy.get('.leaflet-marker-icon').should('be.visible');
```

Comment:
I'd stick to data attributes. The key is not to use a handle on an element that could even possibly be used for implementation purposes. This prevents brittle tests.

Reply:
Ok- I can live with this for now. But we need to come back to this test fairly soon.